### ISSUE TITLE
Encountered error "status code: 422" many times while importing data.

### ISSUE THREAD

**Issue by wyzhangyuhan** [weaviate/weaviate-python-client#263]
I tried to use weaviate to construct inverted indices of tens of millions of levels of corpus. Just for keyword retrieval.

But I encountered the following error SO many times in the process of importing text corpus after loaded about millions items:
`weaviate.exceptions.UnexpectedStatusCodeException: Create objects in batch! Unexpected status code: 422, with response body: {'error': [{'message': "invalid param 'objects': cannot be empty, need at least one object for batching"}]}. `

I can make sure that the machine resources are sufficient and the data is successfully read and entered into the function ` client.batch.add_data_object`.

I wonder if i use wrong methods of importing or there's something wrong in python client.

Here my code of importing:
```
    def import_as_batch(self, bs, data, id_list):
        print(len(data), len(id_list))
        self.client.batch.configure(
            # `batch_size` takes an `int` value to enable auto-batching
            # (`None` is used for manual batching)
            batch_size=bs,
            # dynamically update the `batch_size` based on import speed
            dynamic=True,
            # `timeout_retries` takes an `int` value to retry on time outs
            timeout_retries=9,
            # checks for batch-item creation errors
            # this is the default in weaviate-client >= 3.6.0
            callback=weaviate.util.check_batch_result,
            consistency_level=weaviate.data.replication.ConsistencyLevel.ALL,  # default QUORUM
            )

        with self.client.batch as batch:
            batch.batch_size = bs
            for idx, d in enumerate(data):
                self.client.batch.add_data_object(d, self.table_name, id_list[idx])
```

And my weaviate version is 1.17.5. Hope can help me analyze the problem, or tell me the direction of investigation. Thanks a lot.

**Comment by dirkkul** [comment]
This sounds like a client problem. Do you have  a lot of timeout errors?

**Comment by wyzhangyuhan** [comment]
Yes!  When this error occurs, it is always accompanied by a timeout retry for the current batch.

After the amount of data in the class reaches close to a million, the error will occur every few tens of thousands of imports.  And I can continue importing after I restart my python program immeadiately but soon comes the error. 

In addition, I once tried to skip the batch with the error using `try-catch` statement and found that none of the subsequent data could be successfully imported.

**Comment by dirkkul** [comment]
Hi @wyzhangyuhan,

where you able to test the version I posted?

**Comment by wyzhangyuhan** [comment]
Thanks for your timely repair! 👍 

I've tried the version you posted the other day. It seems that my data was imported normally and no error was reported. 

During this time,  I tried calling `with_meta_count()` to check the total number and found that the number was not updated synchronously. 

Because I was busy with something else these days, I didn't have time to check the problem carefully. But judging from the final result, my data might have been successfully imported.

My only explanation for the above phenomenon:
- The data entry I enter has its own unique id. 
- During the previous import with old client version, some data was successfully imported despite the 422 error. 
- Therefore, the data entries already exist when using new version, so the total does not change.  :D
